### ISSUE TITLE
[iOS] Can't find buttons by their accessibilityIdentifier in UI Tests

### ISSUE THREAD

**Issue by malaba** [mapbox/mapbox-gl-native#14460]
We are having a problem with our UI Test in our application. Unfortunately if we make a minimal demo app to reproduce the bug it is alway working.

Everything work on version 4.7.1 (and before), but not in version 4.8.0, 4.9.0 and 4.10.0

Basically the ui tests are not able to find the buttons (that are the annotations) on the map. On a version that work, in the debugger if we `po app.buttons` (where `app` is the `XCUIApplication`) we get a list of buttons like:

```
Find: Target Application 'com.blank.demo.development'
  Output: {
    Application, pid: 53012, label: 'Demo'
  }
  ↪︎Find: Descendants matching type Button
    Output: {
      Button, {{16.0, 633.0}, {16.0, 18.0}}, label: 'info small'
      Button, {{16.0, 20.0}, {44.0, 44.0}}, identifier: 'navigation/button/menu', label: 'menu'
      Button, {{327.0, 28.0}, {40.0, 40.0}}, label: 'Compass', value: north
      Button, {{-15.0, -15.0}, {30.0, 30.0}}, label: 'You Are Here', value: 47 degrees and 23 ...
      Button, {{172.0, 324.5}, {32.0, 32.0}}, identifier: 'map/annotations/vehicle-197'
      Button, {{178.5, 318.5}, {32.0, 32.0}}, identifier: 'map/annotations/vehicle-107'
      ...
      }
```
And we can search a buttons by its `accessibilityIdentifier` = "map/annotations/vehicle-107" for example, click on it and continue the test.

But on a version that doesn't work we get:
```
expression produced error:
error: /var/folders/5j/xw7m_dc55k5_z4psl1n3dx2h0000gn/T/expr57-a83ac9..swift:1:72:
error: 'XCUIElementQuery' is not a member type of 'XCTest'
	Swift._DebuggerSupport.stringForPrintObject(Swift.UnsafePointer<XCTest.XCUIElementQuery>(bitPattern: 0x117cf79e0)!.pointee)
```
To investigate we cloned this repo, and build from source many binaries. By doing a manual `git-bisect` we found the exact commit that break our application: fca084b92322c5915d3361ef21c6ea80ed252aed. Before it work, at this commit and thereafter it doesn't.

There is not that much at this commit. So we investigated the code. What we found is that if we "rollback" those change on a recent version (say 4.10) then our ui tests work again!

By rollback we mean, in file `MGLStyle.mm`:
- removing in method `placeStyleLayers()` the two added string: `@"natural_label", @"transit_stop_label"`
- or removing in method `roadStyleLayers()` the string `@"road"` (here we don't need to go back to the `isEqualToString:` way it was before)

So if we do either step, compile and put this `Mapbox` binary in our App, then our UI Tests works again. _We tested this "rollback" on versions 4.8.0, 4.9.0 and 4.10.0. Also on latest `master` on date 2019/04/04 and 2019/04/18. All working with this patch._

We analysed further and also found that those two previous methods `placeStyleLayers()` and `roadStyleLayers()` are only used in `MGLMapView.mm` by `visiblePlaceFeatures()` and `visibleRoadFeatures()` respectively. Those last two methods are then used by `accessibilityValue()`. Here we think we are approaching something since our problem is with those `accessibilityIdentifier`.

Is that report give any idea to someone ?

### Steps to reproduce

I can't produce a demo app that show the bug, it is always working. And I can't give the real app.

### Expected behavior

The `accessibilityIdentifier` should be visible.

### Actual behavior

None of the buttons are available by their `accessibilityIdentifier`

### Configuration

**Mapbox SDK versions:** 4.7.1 to work, 4.8.0/4.9.0/4.10.0 to break our UI Tests
**iOS/macOS versions:** iOS 12.1 and 12.2
**Device/simulator models:** Simulator iPhone 6,7,8
**Xcode version:** 10.1 and 10.2

Thanks for any possible hints,
Pascal

**Comment by jmkiley** [comment]
Hi @malaba -

Thank you for your detailed bug report. Some follow-up questions:

- Could you share which style you are using?  
- Do your tests work as expected in v4.7.0? It looks like https://github.com/mapbox/mapbox-gl-native/commit/fca084b92322c5915d3361ef21c6ea80ed252aed landed in v4.7.0-beta.2, but https://github.com/mapbox/mapbox-gl-native/pull/13646 reverted the use of v11 styles by default in v4.7.1. 
- Are you using `UIButton` objects within the annotation views?

**Comment by malaba** [comment]
Hi @jmkiley, thanks for responding.

Here are your infos:

- The style was private, but I just changed it to some built in style and it's the same exact problem. For example now to test I use `styleURL` = `MGLStyle.outdoorsStyleURL` or `MGLStyle.darkStyleURL`. Both fail the same way as our private style.
- in v4.7.0 it work like a charm (in all style).
- No, we are not using `UIButton` in our subclasses of `MGLAnnotationView`

**Comment by malaba** [comment]
I updated to v5.2.0 and it seems to work now. Don't know why.

We still have issue on some tests with method `accessibilityElementAtIndex()` from `MGLMapView.mm`. It sort all annotations on main thread, take hundred MB of ram (sometimes GB of ram) and pin the CPU at 99.9% for many minutes... hence the application doesn't respond to touch events and UI tests fails by timeout essentially.

If it ring a bell to someone.

**Comment by malaba** [comment]
Again if I remove ` @"road"` from method ` roadStyleLayers()` in `MGLStyle.mm` everything run smooth and fast, no CPU at 99.9% (on source code v5.2, manual built binary).

Also if I comment in `MGLMapView.mm` lines 2742-2787 (basically removing any "sort" around Visible place/road features it work and no CPU at 99.9% ...

What is the problem with "road" ? Is it possible that in a city there is just too many roads and this method take too much time for UI Tests ?

**Comment by friedbunny** [comment]
@malaba glad to hear that this issue appears to have been resolved. We’re tracking the testing performance issue you noted in https://github.com/mapbox/mapbox-gl-native/issues/14324, so I’m going to go ahead and close this issue — thanks for following up. 🙇
